### ISSUE TITLE
Why is there no hover dropdown menu?

### ISSUE THREAD

**Issue by iwasherefirst2** [twbs/bootstrap#27166]
I was a bit surprised that I did not found a hover dropdown menu in Bootstrap. 

Most websites that I know have a dropdown navigation that appears on hover on a desktop PC like this:

![image](https://user-images.githubusercontent.com/1765602/44944713-2c278d00-addc-11e8-9b83-3a3abd65e859.png)

This is also the case for the [#1 Selling HTML Template](https://themeforest.net/item/porto-responsive-html5-template/4106987?s_rank=1) on themeforest.net that buils upon Bootstrap, or allmost all Wordpress themes, like fore example here: http://www2.cuny.edu

The only thing that I found in Bootstrap were [dropdowns for tabs or pills on click](https://getbootstrap.com/docs/4.0/components/navs/#using-dropdowns) (not on hover).

I wonder, is this on purpose? Is there a reason Bootstrap doesn't want to support hover-navigation's that have been around for years all over the web?

**Comment by sts-ryan-holton** [comment]
@iwasherefirst2  I think the main issue with using a hover dropdown rather than, or an alternative to a click is simply because you could have issues in hovering on mobiles (usually where most of the traffic comes from), thus you'd likely have to implement additional functionality or some CSS wizardry to get this to work.

That being said, I think there should be an option as a SCSS variable or via a class which enables hovering over dropdowns and navbar dropdowns

**Comment by patrickhlauke** [comment]
noting here that pure CSS dropdowns won't work for keyboard users (unless we use a solution including `:focus-within` and relevant polyfills for older browsers), and that looking ahead for WCAG 2.1 SC 1.4.13 there needs to be a way for the menu to be closed without moving mouse/focus (which, pretty much, invalidates use of CSS only, as it'll need to provide a JS way of opening and closing on, say, `ESC`)

**Comment by iwasherefirst2** [comment]
@patrickhlauke sure one should use JS to enable keyboard access, closing without mouse, and using the hover. A CSS only solution would not be satisfying.

**Comment by mdo** [comment]
> I wonder, is this on purpose? Is there a reason Bootstrap doesn't want to support hover-navigation's that have been around for years all over the web?

Yes, and yes. There are some old issues around this, and a very old blog post from myself as well: http://markdotto.com/2012/02/27/bootstrap-explained-dropdowns/. I could see allowing it in the future, but as @patrickhlauke, it'd definitely need to be supplemented with JS. Not on the priority list right now though.

**Comment by herbalite** [comment]
Because Bootstap 4 uses jQuery, things that already can be activated through Javascript events, can be combined with the jQuery hoverIntent plugin. https://github.com/briancherne/jquery-hoverIntent

I just added this to a project. And it works fine, with Bootstrap JS events.

The good thing IMO is that it doesn't interfere with any of the functionality provided by Bootstrap, including accessibility. 
Care needs to be taken to not overdue it and apply it to everything within Bootstrap that has JS events.

As for some code example. If interested, that needs to be probably should be done at SO.

**Comment by JPustkuchen** [comment]
@mdo should this perhaps be reopened for bs5? Googling for a navbar desktop hiver solution, it seems that there are pretty many workaround and even business concepts (funny) out there for hover menus.

My point is, that people seem to create their custom solutions to this still typical functionality (for mouse devices) but are happy if it works, without considering WCAG / aria. So it might be better to at least document a best practice, if not providing one in Bootstrap?

If you see the point, should we open a separate issue for that?
